Dart/src/com/jetbrains/lang/dart/analyzer/DartAnalysisServerService.java

```diff
@@ -94,9 +100,23 @@ public void computedErrors(@NotNull final String file, @NotNull final List<Analy
     }
 
     @Override
-    public void computedNavigation(String file, List<NavigationRegion> targets) {
+    public void computedNavigation(String file, List<NavigationRegion> regions) {
       if (DartResolver.USE_SERVER) {
-        myNavigationData.put(FileUtil.toSystemIndependentName(file), targets);
+        file = FileUtil.toSystemIndependentName(file);
+        // Ignore notifications for files that has been changed, but server does not know about it yet.
+        if (myFilePathsWithUnsentChanges.contains(file)) {
+          return;
+        }
+        // Convert NavigationRegion(s) into PluginNavigationRegion(s).
+        List<PluginNavigationRegion> pluginRegions = new ArrayList<PluginNavigationRegion>(regions.size());
+        for (NavigationRegion region : regions) {
+          pluginRegions.add(new PluginNavigationRegion(region));
+        }
+        // Put PluginNavigationRegion(s).
+        synchronized (myLock) {
```

Comment:
Separate pull request or just push it yourself (though to do it you need to have another Community+Dart project configured, based on original intellij-plugins repo)